desktop_version/src/Input.cpp

```diff
@@ -35,6 +35,37 @@ static void updatebuttonmappings(int bind)
     ) {
         if (key.isDown(i))
         {
+            if (!game.gpmenu_confirming || i != game.gpmenu_lastbutton)
+            {
+                game.gpmenu_confirming = true;
+                game.gpmenu_lastbutton = i;
+
+                // Is this button already in the list for this action?
+                std::vector<SDL_GameControllerButton>* vec = NULL;
+                switch (bind)
+                {
+                case 1: vec = &game.controllerButton_flip; break;
+                case 2: vec = &game.controllerButton_map; break;
+                case 3: vec = &game.controllerButton_esc; break;
+                case 4: vec = &game.controllerButton_restart; break;
+                case 5: vec = &game.controllerButton_interact; break;
+                default: return;
```

Comment:
Is this only possible when a manually edited config has an invalid value? If so it may be worth warning just in case someone's messing with this stuff and wants to know what might be wrong.

Reply:
It's only if an unexpected menu option is selected - `bind` comes from the current menu option and this function is only called when it's between 1 and 5.